c2corg_api/views/document.py

```diff
@@ -125,18 +144,25 @@ def create_response():
                 # request equals the current etag, return 'NotModified'
                 etag_cache(self.request, cache_key)
 
-                return get_or_create(
-                    cache_document_detail, cache_key, create_response)
+                return get_or_create(cache, cache_key, create_response)
 
         # don't cache if requesting a document for editing
         return create_response()
 
     def _get_in_lang(self, id, lang, clazz, schema, editing_view,
                      clazz_locale=None, adapt_schema=None,
                      include_maps=True, include_areas=True,
-                     set_custom_associations=None, set_custom_fields=None):
-        document = self._get_document(
-            clazz, id, clazz_locale=clazz_locale, lang=lang)
+                     set_custom_associations=None, set_custom_fields=None,
+                     cook_locale=False):
+
+        if cook_locale:
+            document = self._get_document_for_cooking(
```

Comment:
it's not the same function because `_get_document` returns the expected locale, or no locale if this lang does not exists, where `_get_document_for_cooking` returns the expected locale, or another one, based on lang_priority array.